crates/amaru-consensus/src/consensus/tx_submission/tests/assertions.rs

```diff
@@ -0,0 +1,114 @@
+use pallas_network::miniprotocols::txsubmission::{EraTxBody, EraTxId, Message, TxIdAndSize};
+use pallas_primitives::conway::Tx;
+use std::sync::Arc;
+use std::time::Duration;
+use tokio::sync::mpsc::Receiver;
+use tokio::time::{sleep, timeout};
+
+/// Check that all the given transactions are eventually present in the server mempool.
+#[track_caller]
+pub async fn expect_server_transactions(txs: Vec<Tx>, node_handle: &NodeHandle) {
+    let server_mempool: Arc<dyn Mempool<Tx>> = node_handle.server_mempool.clone();
+    let tx_ids: Vec<_> = txs.iter().map(TxId::from).collect();
+    let mut actual: Vec<_> = tx_ids
+        .iter()
+        .filter(|tx_id| server_mempool.contains(tx_id))
+        .collect();
+
+    let _ = timeout(Duration::from_secs(1000), async {
+        loop {
+            actual = tx_ids
+                .iter()
+                .filter(|tx_id| server_mempool.contains(tx_id))
+                .collect();
+            let all_present = actual.len() == txs.len();
+            if all_present {
+                break;
+            }
+            sleep(Duration::from_millis(10)).await;
+        }
+    })
+    .await;
+    if actual.len() != txs.len() {
+        panic!(
+            "actual transactions\n{}\nexpected transactions\n{}\n",
+            actual
+                .iter()
+                .map(|id| format!("{}", id))
+                .collect::<Vec<_>>()
+                .join(", "),
+            tx_ids
+                .iter()
+                .map(|id| format!("{}", id))
+                .collect::<Vec<_>>()
+                .join(", ")
+        );
+    }
+}
+
+/// Check that the next message is a ReplyTxIds with the expected ids.
+#[track_caller]
+pub async fn assert_tx_ids_reply(
+    rx_messages: &mut Receiver<Message<EraTxId, EraTxBody>>,
+    tx_ids: &[TxId],
+    expected_ids: &[usize],
+) -> anyhow::Result<()> {
+    let era_tx_ids = era_tx_ids(tx_ids);
+    let tx_ids_and_sizes: Vec<TxIdAndSize<EraTxId>> = expected_ids
+        .iter()
+        .map(|&i| TxIdAndSize(era_tx_ids[i].clone(), 50))
+        .collect();
+    assert_next_message(rx_messages, ReplyTxIds(tx_ids_and_sizes)).await?;
+    Ok(())
+}
+
+/// Check that the next message is a ReplyTxs with the expected transaction bodies.
+#[track_caller]
+pub async fn assert_tx_bodies_reply(
+    rx_messages: &mut Receiver<Message<EraTxId, EraTxBody>>,
+    txs: &[Tx],
+    expected_body_ids: &[usize],
+) -> anyhow::Result<()> {
+    let txs: Vec<EraTxBody> = expected_body_ids
+        .iter()
+        .map(|&i| era_tx_body(&txs[i]))
+        .collect();
+
+    assert_next_message(rx_messages, ReplyTxs(txs)).await?;
+    Ok(())
+}
+
+/// Check that the next message matches the expected one.
+#[track_caller]
+pub async fn assert_next_message(
+    rx_messages: &mut Receiver<Message<EraTxId, EraTxBody>>,
+    expected: Message<EraTxId, EraTxBody>,
+) -> anyhow::Result<()> {
+    let actual = rx_messages
```

Comment:
there should be a timeout here so that a test failure can be more easily located